### DISCUSSION TITLE
Intersection Observer problem with Next.js

### DISCUSSION THREAD

**Discussion by elgatoflaco** [vercel/next.js#13866] [Help]
I can't get intersection observer to work in next.js. Any suggestion?

The code I use is as follows.

In create-react-app it works.
In Next.js it returns the following error.

Unhandled Runtime Error
TypeError: Failed to execute 'observe' on 'IntersectionObserver': parameter 1 is not of type 'Element'.
```jsx
useEffect(() => {
    if (typeof window !== "undefined" && typeof document !== "undefined") {
      const headerEl = document.querySelector(".header");
      const sentinalEl = document.querySelector(".sentinal");
      console.log(sentinalEl);
      const handler = (entries) => {
        console.log(entries);
        // entries is an array of observed dom nodes
        // we're only interested in the first one at [0]
        // because that's our .sentinal node.
        // Here observe whether or not that node is in the viewport
        if (!entries[0].isIntersecting) {
          headerEl.classList.add("enabled");
        } else {
          headerEl.classList.remove("enabled");
        }
      };
      // create the observer
      const observer = new window.IntersectionObserver(handler);
      // give the observer some dom nodes to keep an eye on
      observer.observe(sentinalEl);
      console.log(ref.current);
    }
  });
```

I am new to Next.js, I would appreciate any help.

Thanks!!

**Comment by jsbosnic** [comment]
Hey! 
Even though you asked this 1 year ago, I just stumbled upon the same error message.
In my case, I was passing down an object (via useRef) instead of a DOM Node (via ref.current).
It could be that, or the element just doesn't exist at the time you start observing.

Saludos colega!

**Comment by deyvisonborges** [comment]
use ref in your page

**Comment by joonk025** [comment]
I had to put `' observer.observe(sentinalEl);'` inside of a setTimeout
